core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -172,10 +172,11 @@ void *impl_allocate_common(const Cuda &exec_space, const char *arg_label,
     if (exec_space_provided) {
       cudaStream_t stream = exec_space.cuda_stream();
       error_code          = cudaMallocAsync(&ptr, arg_alloc_size, stream);
-      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamSynchronize(stream));
+      exec_space.fence("Kokkos::Cuda: backend fence after async malloc");
     } else {
       error_code = cudaMallocAsync(&ptr, arg_alloc_size, 0);
-      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaDeviceSynchronize());
+      Impl::cuda_device_synchronize(
+          "Kokkos::Cuda: backend fence after async malloc");
```

Comment:
@masterleinad This is just a comment about the `clang-format`. Do you think it would make sense (not in this PR of course) to lower the penalty of excess characters? (See https://clang.llvm.org/docs/ClangFormatStyleOptions.html#penaltyexcesscharacter)

I mean, sometimes a line with 87 characters might be nicer than the same line split weirdly across 2 lines :smile: This is just a question, not suggesting to do anything but gathering your opinion on this :smile:

Reply:
We need to update the clang-format version we are using to something more recent in the near future.
This will trigger changes throughout the code base.  When we do so would be a time where we would consider changing the clang-format configuration.  If that's something you are interested in you are welcome to explore options but this needs to be done in an issue elsewhere.
A word of warning, this is a time sink and at times can get controversial.

Reply:
* If I'm not mistaken, the [PenaltyExcessCharacter](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#penaltyexcesscharacter) is available from `clang-format` 3.7, and you're requiring version `8.0` in `scripts/apply-clang-format`. So I guess the option could be readily used.
* Instead of modifying the whole code base when the clang format changes (either version or options), I think you should use `git-clang-format` to check for style only in modified files. This would avoid committing purely stylistic changes. That would avoid messing up the git history.

Reply:
We allow disabling clang-format when appropriate.  You will find all sort of examples throughout the code base such as https://github.com/kokkos/kokkos/blob/4b6d971dce856132961b1f178dcae1256b92b733/core/src/impl/Kokkos_QuadPrecisionMath.hpp#L46-L67 or https://github.com/kokkos/kokkos/blob/4b6d971dce856132961b1f178dcae1256b92b733/core/src/OpenACC/Kokkos_OpenACC_ParallelFor_Range.hpp#L35-L37

In principle you are right but in practice I think it would mean that the entire file would be reformatted as soon as someone touches it and this "spreads" the noise across the history of the repo instead of having one clean "break".